app/services/katello/event_queue.rb

```diff
@@ -0,0 +1,38 @@
+module Katello
+  class EventQueue
+    @event_types = {}
+
+    def self.clear_events(event_type, object_id, on_or_earlier_than)
+      Katello::Event.where(:in_progress => true, :object_id => object_id, :event_type => event_type).where('created_at <= ?', on_or_earlier_than).destroy_all
+    end
+
+    def self.peek_event
```

Comment:
but its not popping it off the top.  to me peek is a better term than pop, but i can see how both are incorrect.  'grab_task' ?